SECURITY.md

```diff
@@ -0,0 +1,66 @@
+# Security Policy
+
+ - [**Using AutoGPT Securely**](#using-AutoGPT-securely)
+   - [Restrict Workspace](#restrict-workspace)
+   - [Untrusted inputs](#untrusted-inputs)
+   - [Data privacy](#data-privacy)
+   - [Untrusted environments or networks](#untrusted-environments-or-networks)
+   - [Multi-Tenant environments](#multi-tenant-environments)
+ - [**Reporting a Vulnerability**](#reporting-a-vulnerability)
+
+## Using AutoGPT Securely
+
+### Restrict Workspace
+
+Since agents can read and write files, it is important to keep them restricted to a specific workspace. This happens by default *unless* RESTRICT_TO_WORKSPACE is set to False.
+
+We do not recommend disabling RESTRICT_TO_WORKSPACE, unless AutoGPT is running in a sandbox environment where it couldn't do any damage (e.g. Docker or a VM).
+
+### Untrusted inputs
+
+When handling untrusted inputs, it's crucial to isolate the execution and carefully pre-process inputs to mitigate script injection risks.
+
+For maximum security when handling untrusted inputs, you may need to employ the following:
+
+* Sandboxing: Isolate the process.
+* Updates: Keep your libraries (including AutoGPT) updated with the latest security patches.
+* Input Sanitation: Before feeding data to the model, sanitize inputs rigorously. This involves techniques such as:
+    * Validation: Enforce strict rules on allowed characters and data types.
+    * Filtering: Remove potentially malicious scripts or code fragments.
+    * Encoding: Convert special characters into safe representations.
+    * Verification: Run tooling that identifies potential script injections (e.g. [models that detect prompt injection attempts](https://python.langchain.com/docs/guides/safety/hugging_face_prompt_injection)). 
+
+### Data privacy
+
+To protect sensitive data from potential leaks or unauthorized access, it is crucial to sandbox the agent execution. This means running it in a secure, isolated environment, which helps mitigate many attack vectors.
+
+### Untrusted environments or networks
+
+Since AutoGPT performs network calls to the OpenAI API, it is important to always run it with trusted environments and networks. Running it on untrusted environments can expose your API KEY to attackers.
+Additionally, running it on an untrusted network can expose your data to potential network attacks. 
+
+However, even when running on trusted networks, it is important to always encrypt sensitive data while sending it over the network.
+
+### Multi-Tenant environments
+
+If you intend to run multiple AutoGPT brains in parallel, it is your responsibility to ensure the models do not interact or access each other's data. Besides beware that if the instances are using the same API KEY, they will share the same limits.
```

Comment:
Unclear on exactly what is meant by this.

Reply:
Yeah, that's the idea. Since users might want to try using AutoGPT in parallel (with co-agents running at the same time) it is their responsability to ensure isolation, resource allocation and other security aspects.

Reply:
Okay, thanks for the clarification.

> Besides beware that if the instances are using the same API KEY, they will share the same limits.

This part doesn't seems relevant to a security file if we're talking about rate limits, and I don't want to imply people should try to bypass a provider's rate limits by having multiple accounts.

Apologies if I've misunderstood.

Reply:
> Besides beware that if the instances are using the same API KEY, they will share the same limits.

Yeah, you are right, it might lead to the understanding that they should bypass the rate limit -- my idea was to focus on preventing denial of service attacks that would attack a tenant to hit the rate limit and leave other tenants unavailable, but this is generically mentioned in "Resource Allocation" so it is better to just remove this part.